docs/platforms/javascript/common/enriching-events/transaction-name/index.mdx

```diff
@@ -22,15 +40,104 @@ task being executed. For example:
+
+```javascript
+const activeSpan = Sentry.getActiveSpan();
+const rootSpan = activeSpan && Sentry.getRootSpan(activeSpan);
+
+Sentry.updateSpanName(rootSpan, "UserListView");
+```
+
+Learn more about <PlatformLink to="/tracing/instrumentation/custom-instrumentation/#updating-the-span-name">updating the span name</PlatformLink>.
+
+### After the root span finished
+
+If you cannot determine the root span name before the span finishes, you can retroactively change it in `beforeSendTransaction`:
+
+```javascript
+Sentry.init({
+  beforeSendTransaction(event) {
+    event.transaction = "UserListView";
+    return event;
+  },
+});
+```
 
-A lot of our framework integrations already set a transaction name, though you can set one yourself.
+We recommend to only do this, if you cannot use any of the other options to set the root span name earlier.
 
-You'll first need to import the SDK, as usual:
+## Further Information
 
-<PlatformContent includePath="enriching-events/import" />
+You might be wondering why these two types of transaction names exist and why they are set independently.
+The reason is mostly historic evolvement of the Sentry product and the SDKs.
+The error transaction name existed long before Sentry provided tracing capabilities and where it was only applied to error events to better group issues.
 
-To override the name of the currently running transaction:
+In the first iteration of Sentry's tracing or performance monitoring product, we decided to call the root span of a span tree within an application a "Transaction", which also had a name.
+In the second, now ongoing, tracing iteration, Sentry's new UI features no longer mention "Transactions" for root spans but rather just "spans" and "traces".
+However, the two concepts (error transaction name and root span name) are still ambiguous and used so throughout various older parts of the Sentry UI.
 
-<PlatformContent includePath="enriching-events/set-transaction-name" />
+With this product evolvement in mind, the SDKs also had to adapt their APIs around tracing, specifically moving away from transaction-based APIs in version 8.
+Consequently, the SDK exposes the `setTransactionName` API on the `Scope` which—although the name might suggest differently from historic context—has nothing to do with spans.
+Likewise, the span name is not automatically applied to error events either, meaning both concepts are completely [decoupled](https://github.com/getsentry/sentry-javascript/issues/10846) from each other.
 
-Please refer to [the tracing documentation](../../tracing/) for how to start and stop transactions.
+We believe that the ambiguity of these APIs will decrease over time as the Sentry product further moves away from associating the "Transaction" term with tracing and spans.
```

Comment:
```suggestion
As Sentry continues to evolve, the ambiguity between these terms is expected to decrease as the product moves away from associating "transactions" with tracing and spans.
```
I understand why you added it, but I'm not sure this section is necessary. I think it might add more confusion. But let me know if you disagree :)